### PR TITLE
Fix proxy middleware

### PR DESCRIPTION
**Pull request by kmteras**

### CODE REVIEW

File: lib/salemove/http_client/middleware/proxy.ex [hunk 1]
```diff
@@ -17,7 +17,9 @@ defmodule Salemove.HttpClient.Middleware.Proxy do
       |> inject_proxy(opts)
       |> Tesla.run(next)
     else
-      Tesla.run(env, next)
+      env
+      |> inject_adapter(opts)
+      |> Tesla.run(next)
```
Comment (oskarkook) [inline lib/salemove/http_client/middleware/proxy.ex:22]: Is this covered with a test now?

Comment (kmteras) [reply lib/salemove/http_client/middleware/proxy.ex:22]: The "when client disabled proxy on module level" test kind of covers it as it accesses the adapter options and would have previously failed.

Can add a more specific test as well if you find it necessary.

File: lib/salemove/http_client/middleware/proxy.ex [hunk 2]
```diff
@@ -42,13 +44,23 @@ defmodule Salemove.HttpClient.Middleware.Proxy do
       |> Keyword.put(:proxy, proxy)
 
     new_options =
-      Keyword.update(opts, :adapter_options, proxy_options, fn adapter_options ->
+      opts
+      |> adapter_options()
+      |> Keyword.update(:__adapter_options, proxy_options, fn adapter_options ->
```
Comment (kmteras) [inline lib/salemove/http_client/middleware/proxy.ex:49]: @take-five did the whole `opts` of the proxy have to be put into the env or should the proxy also work with just the `adapter_options` (as it is part of those options)?

File: lib/salemove/http_client/middleware/proxy.ex (hunk 1 above)
Comment (oskarkook) [reply lib/salemove/http_client/middleware/proxy.ex:22]: Add a specific test that expects the options to always be set.

Comment (oskarkook) [reply lib/salemove/http_client/middleware/proxy.ex:22]: Would be good to have to avoid regressions in my opinion.

**Review by oskarkook — APPROVED** [review]

**Review by take-five — APPROVED** [review]

**Comment by kmteras** [comment]
!squash

**Comment by kmteras** [comment]
!merge

**Pull request merged**